client/__tests__/e2e/e2e.test.ts

```diff
@@ -85,448 +78,477 @@ const setToRemoveFrom = "empty_this_geneset";
-      if (i !== 0) await clickOn(`colorby-${label}`);
-      const generatedLabels = await getAllByClass("centroid-label");
+      if (i !== 0) await page.getByTestId(`colorby-${label}`).click();
+      const generatedLabels = await page.getByTestId("centroid-label").all();
+
       // Number of labels generated should be equal to size of the object
       expect(generatedLabels).toHaveLength(
-        // @ts-expect-error ts-migrate(7053) FIXME: Element implicitly has an 'any' type because expre... Remove this comment to see the full error message
         Object.keys(data.categorical[label]).length
       );
     }
   });
 });
 
 describe("graph overlay", () => {
-  test("transform centroids correctly", async () => {
-    await goToPage(pageUrl);
+  test("transform centroids correctly", async ({ page }) => {
+    await goToPage(page);
+    const category = Object.keys(
+      data.categorical
+    )[0] as keyof typeof data.categorical;
 
-    const category = Object.keys(data.categorical)[0];
+    await page.getByTestId(`colorby-${category}`).click();
+    await page.getByTestId("centroid-label-toggle").click();
+    await page.getByTestId("mode-pan-zoom").click();
 
-    await clickOn(`colorby-${category}`);
-    await clickOn("centroid-label-toggle");
-    await clickOn("mode-pan-zoom");
-
-    const panCoords = await calcTransformDragCoordinates(
+    const panCoords = await calcDragCoordinates(
       "layout-graph",
-      data.pan["coordinates-as-percent"]
+      data.pan["coordinates-as-percent"],
+      page
     );
 
-    // @ts-expect-error ts-migrate(7053) FIXME: Element implicitly has an 'any' type because expre... Remove this comment to see the full error message
     const categoryValue = Object.keys(data.categorical[category])[0];
     const initialCoordinates = await getElementCoordinates(
-      `${categoryValue}-centroid-label`
+      `centroid-label`,
+      categoryValue,
+      page
     );
 
-    await tryUntil(async () => {
-      await drag("layout-graph", panCoords.start, panCoords.end, false);
-
-      const terminalCoordinates = await getElementCoordinates(
-        `${categoryValue}-centroid-label`
-      );
-
-      expect(terminalCoordinates[0] - initialCoordinates[0]).toBeCloseTo(
-        panCoords.end.x - panCoords.start.x
-      );
-      expect(terminalCoordinates[1] - initialCoordinates[1]).toBeCloseTo(
-        panCoords.end.y - panCoords.start.y
-      );
-    });
+    await tryUntil(
+      async () => {
+        await drag("layout-graph", panCoords.start, panCoords.end, page);
+
+        const terminalCoordinates = await getElementCoordinates(
+          `centroid-label`,
+          categoryValue,
+          page
+        );
+
+        expect(terminalCoordinates[0] - initialCoordinates[0]).toBeCloseTo(
+          panCoords.end.x - panCoords.start.x
+        );
+        expect(terminalCoordinates[1] - initialCoordinates[1]).toBeCloseTo(
+          panCoords.end.y - panCoords.start.y
```

Comment:
oh this asserts the dragging. Maybe we can do something similar for the tests above in the new tickets!

Reply:
there aren't any elements to track for the above 😢

Reply:
Oy got it. Chromatic it is 🎉 !